### PR TITLE
SONAR-8878 Add a sorting buttons on projects page facets

### PR DESCRIPTION
**Pull request by gregaubert**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/projects/store/utils.js [hunk 1]
```diff
@@ -55,7 +55,9 @@ export const parseUrlQuery = urlQuery => ({
   'duplications': getAsNumericRating(urlQuery['duplications']),
   'size': getAsNumericRating(urlQuery['size']),
   'languages': getAsArray(urlQuery['languages'], getAsString),
-  'search': getAsString(urlQuery['search'])
+  'search': getAsString(urlQuery['search']),
+  'sortby': getAsString(urlQuery['sortby']),
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/store/utils.js]: I'd prefer just `sort`

File: server/sonar-web/src/main/js/apps/projects/store/utils.js [hunk 2]
```diff
@@ -168,6 +170,40 @@ export const convertToFilter = (query, isFavorite) => {
...
+      return 'duplicated_lines_density';
+    case 'size':
+      return 'ncloc';
+    default:
+      return query.sortby;
+  }
+};
+
+export const convertToQueryData = (query, isFavorite, organization, defaultData = {}) => {
+  const data = Object.assign({}, defaultData);
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/store/utils.js]: `const data = { ...defaultData }`

File: server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  static defaultProps = {
+    sortDesc: 'left',
+    leftText: translate('worst'),
+    rightText: translate('best')
+  };
+
+  isSortActive (side) {
+    if (this.props.query.sortby === this.props.property) {
+      if (['false', 'no'].includes(this.props.query.asc)) {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js]: why `no`?

File: server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    const { leftText, rightText } = this.props;
+
+    return (
+        <div className="projects-facet-sort">
+          <span>{translate('projects.sort_list')}</span>
+          <div className="spacer-left button-group">
+            <Link
+              onClick={this.blurLink}
+              className={this.getLinkClass('left')}
+              to={this.getLinkPath('left')}>{leftText.toLowerCase()}</Link>
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js]: I'd not put `toLowerCase()` here, instead use css for it

File: server/sonar-web/src/main/less/init/forms.less [hunk 5]
```diff
@@ -149,6 +149,17 @@ input[type="submit"].button-success {
   }
 }
 
+.button-grey,
+input[type="submit"].button-grey {
+  border-color: @secondFontColor;
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/less/init/forms.less]: make it a bit lighter, for example `#b4b4b4`

File: server/sonar-web/src/main/js/apps/projects/styles.css [hunk 6]
```diff
@@ -106,6 +106,19 @@
...
 }
 
+.projects-facet-sort {
+  float: right;
+  font-weight: normal;
+  font-size: 11px;
+  color: #777;
+}
+
+.projects-facet-sort .button-small {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/styles.css]: buttons are not aligned with the title

File: server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js (hunk 3 above)
Comment (gregaubert) [inline server/sonar-web/src/main/js/apps/projects/filters/SortingFilter.js]: For compatibility with the api, it accepts `false`, `no`, `true` or `yes`

**Comment by gregaubert** [comment]
@stas-vilchik I changed the way the ascending/descending sorting works and some stuff in `store/utils`. So you might want to review my 2 last commits before I merge.

File: server/sonar-web/src/main/js/apps/projects/store/utils.js [hunk 7]
```diff
@@ -142,64 +141,51 @@ const convertToFilter = (query, isFavorite) => {
...
-      return 'duplicated_lines_density';
-    case 'size':
-      return 'ncloc';
-    default:
-      return query.sortby;
+export const convertToSorting = ({ sort }) => {
+  if (sort && ['-', '+'].includes(sort[0])) {
+    return { s: mapPropertyToMetric(sort.substr(1)), asc: sort[0] !== '-' };
   }
+  return { s: mapPropertyToMetric(sort.trim()) };
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/projects/store/utils.js]: why `trim()` ?

File: server/sonar-web/src/main/js/apps/projects/store/utils.js [hunk 8]
```diff
@@ -69,15 +70,15 @@ const convertIssuesRating = (metric, rating) => {
 const convertCoverage = coverage => {
   switch (coverage) {
     case 1:
-      return 'coverage >= 80';
+      return mapPropertyToMetric('coverage') + ' >= 80';
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/projects/store/utils.js]: BLOCKER Move the declaration of "mapPropertyToMetric" before this usage. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![BLOCKER][BLOCKER] 1 blocker

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'

**Pull request merged**